### PR TITLE
feat(medusa): Update shipping option price type

### LINKED ISSUE #5824: Unable to change shipping price type
## Bug report
I can not change Price Type in Shipping Option from "Flat Rate" to "Calculated".

### Describe the bug
When I am trying to, I am getting notification with success message, but it is not updating. I have to recreate shipping option.
### System information

Medusa version (including plugins): 1.18.0
Node.js version: 18.17.1
Database: postgresql
Operating system: ubuntu 20.04
Browser (if relevant): brave

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- allow updates to shipping option price type
- Clean up the use of strings in the api validation

Fixes #5824

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/shipping-options/create-shipping-option.ts [hunk 1]
```diff
@@ -203,15 +211,16 @@ export class AdminPostShippingOptionsReq {
   @IsString()
   provider_id: string
 
-  @IsOptional()
   @IsString()
-  profile_id?: string
+  profile_id: string
```
Comment (adrien2p) [inline packages/medusa/src/api/routes/admin/shipping-options/create-shipping-option.ts]: todo: Looks like the profile_id is now required, but in the end point if it is not present we are retrieving the default one. So I would suggest the API to not required this property. wdyt?

File: packages/medusa/src/services/shipping-option.ts [hunk 2]
```diff
@@ -75,7 +77,7 @@ class ShippingOptionService extends TransactionBaseService {
    * @return {ShippingOptionRequirement} a validated shipping requirement
    */
   async validateRequirement_(
-    requirement: ShippingOptionRequirement,
+    requirement: validateRequirementTypeInput,
```
Comment (adrien2p) [inline packages/medusa/src/services/shipping-option.ts]: todo: upperCase V for the first letter.

**Review by adrien2p — APPROVED** [review]
LGTM, few todos

File: packages/medusa/src/types/shipping-options.ts [hunk 3]
```diff
@@ -50,7 +51,15 @@ export type CreateShippingOptionInput = {
   is_return?: boolean
   admin_only?: boolean
   metadata?: Record<string, unknown>
-  requirements?: ShippingOptionRequirement[]
+  requirements?: RequirementInput[]
+}
+
+export type RequirementInput = { type: RequirementType; amount: number }
+
+export type validateRequirementTypeInput = {
```
Comment (olivermrbl) [inline packages/medusa/src/types/shipping-options.ts]: todo: capitalize

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**